plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,266 @@
+                            new UpdateColumnStatisticsForTableRequest()
+                                    .withCatalogId(catalogId)
+                                    .withDatabaseName(table.getDatabaseName())
+                                    .withTableName(table.getTableName())
+                                    .withColumnStatisticsList(columnChunk))))
+                    .collect(toUnmodifiableList());
+
+            getFutureValue(allOf(writeChunkFutures.toArray(CompletableFuture[]::new)));
+
+            final Map<String, HiveColumnStatistics> currentTableColumnStatistics = this.getTableColumnStatistics(table);
+            Set<String> removedStatistics = difference(currentTableColumnStatistics.keySet(), updatedTableColumnStatistics.keySet());
+            List<CompletableFuture<Void>> writeFutures = removedStatistics.stream()
+                    .map(column -> runAsync(() ->
+                            glueClient.deleteColumnStatisticsForTable(
+                                    new DeleteColumnStatisticsForTableRequest()
+                                            .withCatalogId(catalogId)
+                                            .withDatabaseName(table.getDatabaseName())
+                                            .withTableName(table.getTableName())
+                                            .withColumnName(column)))
+                    ).collect(toUnmodifiableList());
+
+            for (CompletableFuture<Void> writeFuture : writeFutures) {
+                getFutureValue(writeFuture);
+            }
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public void updatePartitionStatistics(Partition partition, Map<String, HiveColumnStatistics> updatedColumnStatistics)
+    {
+        try {
+            HiveBasicStatistics partitionStats = getHiveBasicStatistics(partition.getParameters());
+            List<ColumnStatistics> columnStats = toGlueColumnStatistics(partition, updatedColumnStatistics, partitionStats.getRowCount());
+
+            List<List<ColumnStatistics>> columnChunks = Lists.partition(columnStats, GLUE_COLUMN_WRITE_STAT_PAGE_SIZE);
+
+            List<CompletableFuture<Void>> writePartitionStatsFutures = columnChunks.stream()
+                    .map(columnChunk ->
+                            runAsync(() -> glueClient.updateColumnStatisticsForPartition(new UpdateColumnStatisticsForPartitionRequest()
+                                    .withCatalogId(catalogId)
+                                    .withDatabaseName(partition.getDatabaseName())
+                                    .withTableName(partition.getTableName())
+                                    .withPartitionValues(partition.getValues())
+                                    .withColumnStatisticsList(columnChunk)), writeExecutor)
+                    ).collect(toUnmodifiableList());
```

Comment:
nit: move closing brackets to previous line. I wonder why checkstyle didn't complain though.